api/v1beta1/validations.go

```diff
@@ -118,6 +123,168 @@ func ValidateCount(count *intstr.IntOrString) error {
+					return fmt.Sprintf("could not determine value %v for field %s", fe.Value(), fe.Field())
+				}
+
+				i = int(unsignedVal)
+			} else {
+				if iPtr == nil {
+					i = 0
+				} else {
+					i = *iPtr
+				}
+			}
+		}
+
+		iStr := strconv.Itoa(i)
+		t, _ := ut.T("gte", fe.Namespace(), iStr, fe.Param())
+
+		return t
+	})
+}
+
+func registerLteTranslation(validate *validator.Validate, translator ut.Translator) error {
+	return validate.RegisterTranslation("lte", translator, func(ut ut.Translator) error {
+		// The {idx} values are interpolated using the arguments to the ut.T("lte", ...) call in the function below
+		return ut.Add("lte", "{0} is set to {1}, but must be less or equal to {2}", true)
+	}, func(ut ut.Translator, fe validator.FieldError) string {
+		// values that are constrained by "lte" can include: int, *int, uint
+		// fe.Value() is an interface{}, so we need to check each of these type options, one by one
+		i, ok := fe.Value().(int)
+		if !ok {
+			iPtr, k := fe.Value().(*int)
+			if !k {
+				unsignedVal, k3 := fe.Value().(uint)
+				if !k3 {
+					// this will be directly seen by the user if their field fails validation.
+					return fmt.Sprintf("could not determine value %v for field %s", fe.Value(), fe.Field())
+				}
+
+				i = int(unsignedVal)
+			} else {
+				if iPtr == nil {
+					i = 0
+				} else {
+					i = *iPtr
+				}
+			}
+		}
+
+		iStr := strconv.Itoa(i)
+		t, _ := ut.T("lte", fe.Namespace(), iStr, fe.Param())
+
+		return t
+	})
+}
+
+func registerRequiredTranslation(validate *validator.Validate, translator ut.Translator) error {
+	return validate.RegisterTranslation("required", translator, func(ut ut.Translator) error {
+		// The {idx} values are interpolated using the arguments to the ut.T("required", ...) call in the function below
+		return ut.Add("required", "{0} is a required field, and must be set", true)
+	}, func(ut ut.Translator, fe validator.FieldError) string {
+		t, _ := ut.T("required", fe.Namespace())
+
+		return t
+	})
+}
+
+func registerOneofciTranslation(validate *validator.Validate, translator ut.Translator) error {
+	return validate.RegisterTranslation("oneofci", translator, func(ut ut.Translator) error {
+		// The {idx} values are interpolated using the arguments to the ut.T("oneofci", ...) call in the function below
+		return ut.Add("oneofci", "{0} is set to {1}, but must be one of the following: \"{2}\"", true)
+	}, func(ut ut.Translator, fe validator.FieldError) string {
+		options := fe.Param()
+
+		userOptionsString := strings.Join(strings.Split(options, " "), ", ")
+
+		userChoiceStr, ok := fe.Value().(string)
+		if !ok {
+			return fmt.Sprintf("could not permit value \"%v\" for field %s, try one of \"%s\"", fe.Value(), fe.Field(), userOptionsString)
+		}
+
+		t, _ := ut.T("oneofci", fe.Namespace(), userChoiceStr, userOptionsString)
+
+		return t
+	})
+}
+
+func ValidateStructTags(s DisruptionSpec) error {
```

Comment:
Does this function needs to be public?

Reply:
Also why not link this function to the struct itself?

Reply:
It didn't need to be public. Instead of linking it to the struct, I replaced the arg with `interface {}` and I'll re-use it identically for DisruptionCrons later